slack-conversation.c

```diff
@@ -35,10 +72,41 @@ static void conversations_list_cb(SlackAccount *sa, gpointer data, json_value *j
 		slack_login_step(sa);
 }
 
+static void get_unread_messages_cb(SlackAccount *sa, gpointer data, json_value *json, const char *error) {
+	json_value *all_ims[2] = { json_get_prop(json, "ims"), json_get_prop(json, "mpims") };
+
+	for (unsigned im_type = 0; im_type < 2; im_type++) {
+		json_value *chans = all_ims[im_type];
+		for (unsigned i = 0; chans && i < chans->u.array.length; i++) {
+			json_value *chan_json = chans->u.array.values[i];
+			const char *latest = json_get_prop_strptr(chan_json, "latest");
+			const char *last_read = json_get_prop_strptr(chan_json, "last_read");
+			if (latest && last_read && strcmp(latest, last_read) == 0) {
```

Comment:
You can use `purple_strequal()` instead, to not need all the null checks (slightly easier to read than `strcmp()`)

Reply:
Sure, I can, but just to check, does `purple_strequal()` return false when both elements are null?

Reply:
Fair enough, no, it'll treat two null-strings as equal, since they're both null.